bool    center_flg
bool    back_flg
bool    left_flg
bool    right_flg